microsoft-edge/web-platform/ad-selection-api.md

```diff
@@ -0,0 +1,300 @@
+
+The Ad Selection API provides different services that need to be deployed by sellers or buyers.
+
+
+<!-- ------------------------------ -->
+#### Images for deploying services
+
+Add your User-Defined Functions within the provided images of services.  The User-Defined Functions can run custom business logic.  The User-Defined Functions run in private containers within the deployed services.
+
+Each image defines one service.  Some images don't contain UDFs, and consist entirely of predefined code.  Some images contain UDFs, as functions with an empty body or starter code, that are the relevant User-Defined Functions for that service.  Fill in the bodies of the User-Defined Functions in these images with your own custom code.
+
+Microsoft provides an image for each service, which can be deployed on a cloud provider.  These are the public images that must be used for deployment.  Only official images from Microsoft are able to run private auctions.
+
+Images for seller services:
+* [SellerFrontEnd service](https://mcr.microsoft.com/product/ad-selection/azure/seller-frontend-service)
+* [Auction service](https://mcr.microsoft.com/product/ad-selection/azure/auction-service)
+
+Images for buyer services:
+* [BuyerFrontEnd service](https://mcr.microsoft.com/product/ad-selection/azure/buyer-frontend-service)
+* [Bidding service](https://mcr.microsoft.com/product/ad-selection/azure/bidding-service)
+* [K-Anonymity service](https://mcr.microsoft.com/product/ad-selection/azure/k-anonymity-service)
+
+
+<!-- ------------------------------ -->
+#### Services for sellers
+
+| Service | Description |
+| --- | --- |
+| **SellerFrontEnd** | Provides a `/SelectAd` gRPC endpoint, which receives requests from the seller's ad service to initiate the Protected Audience auction flow. |
+| **Auction** | Provides a `/ScoreAds` gRPC endpoint, which receives requests from the **SellerFrontEnd** service, containing bids that are participating in the auction.  Responds with a score value that the **SellerFrontEnd** service uses to choose the winner. |
+| **Key/Value** | Receives requests from the **SellerFrontEnd** service, which contain lookup keys from buyers' bids (such as `ad_render_urls`).  Returns real-time scoring signals that are required for the auction.  Runs in Bring Your Own Service (BYOS) mode, so the seller does not need to deploy this service in a trusted execution environment (TEE), and can instead use the image that's provided by Microsoft. |
+
+
+<!-- ------------------------------ -->
+#### Services for buyers
+
+| Service | Description |
+| --- | --- |
+| **BuyerFrontEnd** | Provides a `/GetBids` gRPC endpoint, which receives requests from the **SellerFrontEnd** service to initiate the bidding flow. |
+| **Bidding** | Provides a `/GenerateBids` endpoint, which receives requests from the **BuyerFrontEnd** service to handle the bidding and generate a bid.  Generates a bid, chooses the winner, and selects the banner to be rendered. |
```

Comment:
```suggestion
| **Bidding** | Provides a `/GenerateBids` gRPC endpoint, which receives requests from the **BuyerFrontEnd** service to handle the bidding and generate a bid.  Generates a bid, chooses the winner, and selects the banner to be rendered. |
```
draft Doc lacks "gRPC" here:
![image](https://github.com/user-attachments/assets/5e3ccb06-4c76-4842-9d0d-0eff78a8b160)

Reply:
assumed it's a gPRC endpoint, like the other endpoints.  added "gRPC".
@yanushP